src/MultiDimFit.cc

```diff
@@ -665,42 +680,148 @@ void MultiDimFit::doGrid(RooWorkspace *w, RooAbsReal &nll)
           }
 
             //if (verbose > 1) std::cout << "Point " << i << "/" << points << " " << poiVars_[0]->GetName() << " = " << x << std::endl;
-             std::cout << "Point " << i << "/" << points << " " << poiVars_[0]->GetName() << " = " << x << std::endl;
+            std::cout << "Point " << i << "/" << points << " " << poiVars_[0]->GetName() << " = " << x << std::endl;
+            *params = snap;
+            poiVals_[0] = x;
+            poiVars_[0]->setVal(x);
+
+
+            ///////////////////////////////////////////////////////////////////////////////////////////////////
+            /////////////////// Loop over rand points for each profiled POI to get best nll ///////////////////
+
+            bool ok;
+            int n_prof_params = specifiedVars_.size();
+            std::vector<float> nll_at_alt_start_pts;
+
+            // Get vector of points to try
+            float prof_start_pt_range_max = 20.0; // Is this what we want?
```

Comment:
For a generic version of this algorithm, I think you'd want to potentially have a different range for each of the POIs, maybe the range can be read from the ws for each POI? (or the range set by --setParameterRanges)